.github/workflows/codeql-daily.yml

```diff
@@ -8,35 +8,39 @@ on:
 
 jobs:
   analyze:
+    permissions:
+      actions: read  # for github/codeql-action/init to get workflow details
+      contents: read  # for actions/checkout to fetch code
```

Comment:
does it make sense to move `contents: read` up to the top-level like in the others?